fMRIVolume/scripts/OneStepResampling.sh

```diff
@@ -244,50 +260,93 @@ ${FSLDIR}/bin/imcp ${WD}/${T1wImageFile}.${FinalfMRIResolution} ${fMRIFolder}/${
-  if [ $k -eq 0 ] ; then
-    echo "0" >> ${fMRIFolder}/Movement_RelativeRMS.txt
-  else
-    rmsdiff ${MotionMatrixFolder}/${MotionMatrixPrefix}${vnum} $prevmatrix ${ScoutInputgdc} ${ScoutInputgdc}_mask.nii.gz | tail -n 1 >> ${fMRIFolder}/Movement_RelativeRMS.txt
-  fi
-  prevmatrix="${MotionMatrixFolder}/${MotionMatrixPrefix}${vnum}"
-
-  # Combine GCD with motion correction
-  ${FSLDIR}/bin/convertwarp --relout --rel --ref=${WD}/prevols/vol${vnum}.nii.gz --warp1=${GradientDistortionField} --postmat=${MotionMatrixFolder}/${MotionMatrixPrefix}${vnum} --out=${MotionMatrixFolder}/${MotionMatrixPrefix}${vnum}_gdc_warp.nii.gz
-  # Add in the warp to MNI152
-  ${FSLDIR}/bin/convertwarp --relout --rel --ref=${WD}/${T1wImageFile}.${FinalfMRIResolution} --warp1=${MotionMatrixFolder}/${MotionMatrixPrefix}${vnum}_gdc_warp.nii.gz --warp2=${OutputTransform} --out=${MotionMatrixFolder}/${MotionMatrixPrefix}${vnum}_all_warp.nii.gz
-  # Apply one-step warp, using spline interpolation
-  ${FSLDIR}/bin/applywarp --rel --interp=spline --in=${WD}/prevols/vol${vnum}.nii.gz --warp=${MotionMatrixFolder}/${MotionMatrixPrefix}${vnum}_all_warp.nii.gz --ref=${WD}/${T1wImageFile}.${FinalfMRIResolution} --out=${WD}/postvols/vol${vnum}.nii.gz
 
-  # Generate a mask for keeping track of spatial coverage (use nearest neighbor interpolation here)
-  ${FSLDIR}/bin/fslmaths ${WD}/prevols/vol${vnum}.nii.gz -mul 0 -add 1 ${WD}/prevols/vol${vnum}_mask.nii.gz
-  ${FSLDIR}/bin/applywarp --rel --interp=nn --in=${WD}/prevols/vol${vnum}_mask.nii.gz --warp=${MotionMatrixFolder}/${MotionMatrixPrefix}${vnum}_all_warp.nii.gz --ref=${WD}/${T1wImageFile}.${FinalfMRIResolution} --out=${WD}/postvols/vol${vnum}_mask.nii.gz
+if ((useWbResample))
+then
+    affseries="$(mktemp --tmpdir OneStepResampleAffSeries_XXXXXX.txt)"
+    tempfiles_add "$affseries"
+    
+    for ((k=0; k < $NumFrames; k++))
+    do
+        vnum=`${FSLDIR}/bin/zeropad $k 4`
+        
+        #use unquoted $() to change whitespace to spaces
+        echo $(cat "$MotionMatrixFolder/${MotionMatrixPrefix}$vnum") >> "$affseries"
+
+        # Add stuff for estimating RMS motion
+        rmsdiff ${MotionMatrixFolder}/${MotionMatrixPrefix}${vnum} ${MotionMatrixFolder}/${MotionMatrixPrefix}0000 ${ScoutInputgdc} ${ScoutInputgdc}_mask.nii.gz | tail -n 1 >> ${fMRIFolder}/Movement_AbsoluteRMS.txt
+        if [ $k -eq 0 ] ; then
+            echo "0" >> ${fMRIFolder}/Movement_RelativeRMS.txt
+        else
+            rmsdiff ${MotionMatrixFolder}/${MotionMatrixPrefix}${vnum} $prevmatrix ${ScoutInputgdc} ${ScoutInputgdc}_mask.nii.gz | tail -n 1 >> ${fMRIFolder}/Movement_RelativeRMS.txt
+        fi
+        prevmatrix="${MotionMatrixFolder}/${MotionMatrixPrefix}${vnum}"
+    done
+    
+    #gdc warp space is input to input, affine is input to input, OutputTransform is input to MNI
+    xfmargs=(-warp "$GradientDistortionField.nii.gz" -fnirt "$InputfMRI"
```

Comment:
I prefer ${}, particularly when relying on non text for the parsing into variable/non-variable.  I was confused reading this line.

Reply:
Another option would be `"$GradientDistortionField".nii.gz`.  Quotes do something useful, braces generally do not.

Reply:
Yes that would be clearer too.